taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
+import * as uuid from "uuid";
 import ISO6391 from 'iso-639-1';
 
 /**
  * Sub-component for rendering translation of an "entry"  
 */
-const ListTranslations = ({ nodeObject, setNodeObject }) => {
-    let renderedTranslations = {}
+const ListTranslations = ({ nodeObject, setNodeObject, originalNodeObject }) => {
 
-    Object.keys(nodeObject).forEach((key) => {
+    const [renderedTranslations, setRenderedTranslations] = useState([]) // Stores state of all tags
+    const [mainLang_renderedTranslations, setMainLang_renderedTranslations] = useState([]) // Stores state of main language's tags
+    const [open, setOpen] = useState(false); // Used for Dialog component
+    const [newLanguageCode, setNewLanguageCode] = useState(''); // Used for storing new LC from Dialog
+    const [isValidLanguageCode, setisValidLanguageCode] = useState(false); // Used for validating a new LC
+    const [btnDisabled, setBtnDisabled] = useState(true) // For enabling or disabling Dialog button
+
+    // Helper functions for Dialog component
+    function handleClose() { setOpen(false); }
+    function handleOpen() { setOpen(true); }
+
+    // Used for addition of a translation language
+    function handleAddTranslation(key) {
+        const newRenderedTranslations = [...renderedTranslations, {'languageCode' : key, 'tags' : []}]
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
         
-        // Get all tags and its corresponding language code
-        // Tagids need to be recomputed, so shouldn't be rendered
-        // Main language isn't considered, since it's rendered separately
- 
-        if (key.startsWith('tags') && 
-            !key.endsWith(nodeObject.main_language) && 
-            !key.includes('ids')) {
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject[key] = [];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Used for deleting a translation language
+    function handleDeleteTranslation(key) {
+        const newRenderedTranslations = renderedTranslations.filter(obj => !(key === obj.languageCode))
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
+        
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            delete newNodeObject[key];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Changes the translations to be rendered
+    // Dependent on changes occuring in "originalNodeObject"
+    useEffect(() => {
+
+        // Main langauge tags are considered separately, since they have to be rendered first
+        let mainLangTags = []
+        let otherLangTags = []
```

Comment:
nit: no need to be `let`

Reply:
Since I'm pushing values to it, it has been set as a variable.

Reply:
yep, no need to be `let`. In Js, `let ` is used for reseting the variable. Eg: 
```
let myVar = [1,2]
myVar = [2,3]
```
In the case of `const`, is basically to define a reference to a value and not reset that reference
```
const myConst = []
myConst = [] // this will throw an error
```
but... ` myConst.push[1]` won't throw an error, as you are not reseting to another value but modifying the array itself. This only applies for `[]` or `{}`, not to numbers, booleans, etc.
 
Although you can use them interchangably...